### ISSUE TITLE
Workflows in completed with no failures and < 100% lumis

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#7006]
Jean-Roch reported three workflows that completed without any failures but without 100% lumi sections in the output. There is also this elog: 
https://cms-logbook.cern.ch/elog/Workflow+processing/25081

Taking a look at 
https://cmsweb.cern.ch/reqmgr2/data/request?name=pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564

it has 2 merge failures for RAWAODSIM
```
EXO-RunIISpring16DR80-03516_1MergeRAWAODSIMoutput	success	T1_IT_CNAF	0	1
Job Name: e992eabc-50a3-11e6-a6d0-a0369f23d050-0
WMBS job id: 2503273
Workflow: pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564
Task: /pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564/EXO-RunIISpring16DR80-03516_0/EXO-RunIISpring16DR80-03516_1/EXO-RunIISpring16DR80-03516_1MergeRAWAODSIMoutput
Status: success
Input dataset:
Site: T1_IT_CNAF
Agent: cmssrv217.fnal.gov

EXO-RunIISpring16DR80-03516_1MergeRAWAODSIMoutput	success	T2_US_Caltech	0	1
Job Name: 3fdedeb8-50df-11e6-a6d0-a0369f23d050-1
WMBS job id: 2528135
Workflow: pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564
Task: /pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564/EXO-RunIISpring16DR80-03516_0/EXO-RunIISpring16DR80-03516_1/EXO-RunIISpring16DR80-03516_1MergeRAWAODSIMoutput
Status: success
Input dataset:
Site: T2_US_Caltech
Agent: cmssrv217.fnal.gov
```

which I - initially thought - it would have kicked in the robust merge. I was wrong, the first attempt failed for other reasons and the second attempt managed to merge all files.
The first job merged "events": 418
and the second 'events': 2072,

only these 2 jobs give us a total of 2490 events, different than the workload summary that has 2090 only:
https://cmsweb.cern.ch/couchdb/workloadsummary/_design/WorkloadSummary/_show/histogramByWorkflow/pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564

Still investigating what the problem is, probably have to go deep into the database information... :(

**Comment by amaltaro** [comment]
And in case you want to check it later, I copied the merge tarballs to my afs:
http://amaltaro.web.cern.ch/amaltaro/forAlan/noACDC/

I also jsonized both Reports for the 3273 tarball, just in case.

**Comment by ticoann** [comment]
Hi Alan, where did you see that merge failure I didn't see in the wmstats.

I found something else weird too.
It looks like there are problems with cmssrv217.
Jobs ran fine however those merged files are not linked with block so it wasn’t push to dbs.

You can see successful jobs in cmssrv217.

http://cmssrv217.fnal.gov:5984/wmagent_jobdump%2Fjobs/_design/JobDump/_list/successJobs/statusByWorkflowName?startkey=[%22pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564%22]&endkey=[%22pdmvserv_task_EXO-RunIISpring16DR80-03516__v1_T_160719_140341_8564%22%2C{}]&reduce=false&stale=ok

but 

select block_id from dbsbuffer_file where lfn='/store/mc/RunIISpring16DR80/Higgs0PMToZGTo2QG_width-0p00014_M-4000_13TeV-JHUgen/RAWAODSIM/PUSpring16RAWAODSIM_80X_mcRun2_asymptotic_2016_v3-v1/20000/5A447DC6-9F4F-E611-80B5-0025905B85D0.root’;

returns null.

| block_id |
+----------+
|     NULL |


and no blocks are created for the output dataset.

MariaDB [wmagent]> select * from dbsbuffer_block db inner join dbsbuffer_dataset dd on db.dataset_id = dd.id 
    ->     where dd.path='/Higgs0PMToZGTo2QG_width-0p00014_M-4000_13TeV-JHUgen/RunIISpring16DR80-PUSpring16RAWAODSIM_80X_mcRun2_asymptotic_2016_v3-v1/RAWAODSIM';
Empty set (0.00 sec)

I am not sure why block is not created when file is created initially. I have to dig more.

**Comment by amaltaro** [comment]
Sorry Seangchan. What I wanted to say is that there are two merge jobs that failed in the first attempt, but succeeded in the second one (which could have caused skipped unmerged files).

Yes, I took a look at the dbsbuffer_block yesterday as well and the only sample for this dataset is the DQMIO tier.

I'll check the request parameters and the output map to see if I can figure something.
Thanks for having a look at it!

**Comment by ticoann** [comment]
Hi Alan, The problem seems to be missing algorithm or rather missing entries in dbsbuffer_algo_dataset_assoc table. (There are 24 missing entries and 243 files associated with it)
I am sure how this could happen though.

This is the method which insert the algorithm and association table.

https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/DBS3Buffer/DBSBufferFile.py#L142

And the return value is used for file table insert (dbsbuffer_file.dataset_algo).
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/DBS3Buffer/DBSBufferFile.py#L193

Which means the table entry was created but not committed (or deleted - this is unlikely) . I check the transaction logic but I don't see any thing obviously wrong). All the queries in the create method above should go in the one transaction according to the code.

```
MariaDB [wmagent]> select distinct(df.dataset_algo) from dbsbuffer_file df left outer join dbsbuffer_algo_dataset_assoc ds on df.dataset_algo = ds.id where ds.id is null; 
+--------------+
| dataset_algo |
+--------------+
|         3838 |
|         3835 |
|         3837 |
|         3915 |
|         3910 |
|         3911 |
|         3913 |
|         3912 |
|         3914 |
|         3902 |
|         3908 |
|         3909 |
|         3984 |
|         3985 |
|         3986 |
|         3996 |
|         3987 |
|         3997 |
|         3988 |
|         3998 |
|         3989 |
|         3999 |
|         4138 |
|         4295 |
+--------------+
24 rows in set (0.32 sec)

MariaDB [wmagent]> select count(distinct(df.id)) from dbsbuffer_file df left outer join dbsbuffer_algo_dataset_assoc ds on df.dataset_algo = ds.id where ds.id is null;           
+------------------------+
| count(distinct(df.id)) |
+------------------------+
|                    243 |
+------------------------+
1 row in set (0.37 sec)
```

**Comment by amaltaro** [comment]
It's unlikely, but this agent was facing a very high load in the last days with DB timeouts/deadlocks in a few components. It should not happen, but perhaps this also collaborated with the problem...

**Comment by amaltaro** [comment]
I'm closing it hoping your fix #7010 fixes it :)
